sklearn/cluster/_birch.py

```diff
@@ -392,6 +392,12 @@ class Birch(ClusterMixin, TransformerMixin, BaseEstimator):
         if partial_fit is used instead of fit, they are assigned to the
         last batch of data.
 
+    fit_ : bool
+        True if `fit` method is called.
+
+    partial_fit_ : bool
+        True if `partial_fit` method is called.
+
```

Comment:
I'd say we can make these private probably, @ogrisel do you happen to know if we need to keep these two as public?

Reply:
They are clearly for internal purpose only. We should make them private indeed

Reply:
So do we deprecate their usage while documenting for 2 versions or we just switch to only making them private?

Reply:
As much as I would love to just make them private, I think we need to follow the deprecation period.

Reply:
Usually we do not document deprecated features in the public docstrings. So I would rather be in favor of closing this doc PR and just merging the deprecation PR: #19297 instead.